lesson_02_homework.rb

```diff
@@ -1,46 +1,154 @@
+# variable for title is "Mr.", variable for first_name is "Simon", variable for last_name is "Fuzzyface"
+# then name = Mr. Simon Fuzzyface
     name = title + " " + first_name + " " + last_name
+
+# unless the parameters passed are for title and last_name
+# then name = Mr. Fuzzyface
   elsif title && last_name
     name = title + " " + last_name
+
+# unless the parameters passed are for first_name and last_name
+# then name = Simon Fuzzyface
   elsif first_name && last_name
     name = first_name + " " + last_name
+
+# unless the parameters passed are for first_name
+# then name = Simon
   elsif first_name
     name = first_name
+
+# unless first_name or last_name are not passed
+# raise is used as default for when the necessary variables are not passed, 
+# so we get "Oh no, that doesn't look like a name"
   else
     raise "Oh no, that doesn't look like a name"
   end
-
+# the method has put the values into the appropriate variables
   return name
 end
 
-# Fix Broken Code
+puts full_name("Simon", "Fuzzyface", "Mr.")
+puts full_name("Mr.", "Fuzzyface", "")
+puts full_name("Simon", "Fuzzyface", "")
+puts full_name("Simon", "", "")
 
-def full_name(f, l)
-  puts f + l
-end
 
-puts full_nam("bubba", "chuck")
+# ```
+
+# ### Fix Broken Code
 
+# The following code contains some errors and some coding ettiquette issues. 
+# Fix and refactor the code so that it works and is more clear.
 
-# Coding
+# ```ruby
 
+# This is the original
 
-## Enter your code for your 'add' method below
+# def full_name(f, l)
+#   puts f + l
+# end
+# puts full_nam("bubba", "chuck")
 
 
+# This is the correction
 
+# define method full_name (with 2 parameters)
+# change (f, l) to (first_name, last_name)
+# not technically an error but should have variables that make sense
+def full_name(first_name, last_name)
+# set name to nil
+    name = nil
+# code in where names are passed
+# puts should go after end
+    name = first_name + " " + last_name
+end
 
-## Enter the code for you 'join_strings' method below
+# typo: missing e on full_name
+# names were in wrong order...should be "chuck", "bubba"
+puts full_name("chuck", "bubba")
 
 
+# ```
 
+# ### Coding
 
+# 1. Write a method called `add` that does the following:
+#   - takes two parameters and adds them together returning the result. 
+#   For example `add(1, 2)` should return `3`. 
+#   - Add a line that calls your add function with two values, stores the 
+#   result in a variable, and then prints the result to the screen.
+
+# define method add that has 2 parameters
+def add(num1, num2)
+# add parameters num1 and num2
+  num1 + num2
+end
+# pass value to parameter 1 (num1 = 5), pass value to parameter 2 (num2 = 7), print the added values of num1 and num2 and print
+puts add(5, 7)
+ 
+# define method add that has 2 parameters
+def add(num1, num2)
+# add parameters num1 and num2 and store the value in variable sum
+  sum = num1 + num2
+# print the value of sum
+  puts sum
```

Comment:
Indents were lost with copy/paste. Here it is with indents:
def add(num1 + num2)
  sum = num1 + num2
end
puts add(8, 12)